resources/start_local_mdb.sh

```diff
@@ -52,9 +58,55 @@ start_mdb_and_create_user() {
   ./mongosh test --port $2 --eval "db.createUser({user: '$3', pwd: '$4', roles: ['readWrite']})"
 }
 
+# Usage: start_mdb_with_x509 <type> <port> <mongod dir>
+start_mdb_with_x509() {
+  local type=$1
+  local port=$2
+  local mongod_dir=$3
+  local db_path="${type}_db"
+
+  echo "Starting MongoDB $type with initial configuration on port $port"
+  mkdir -p $db_path
+  $mongod_dir/bin/mongod --dbpath $db_path --port $port &
+
+  echo "Waiting 5 seconds for MongoDB to start..."
+  sleep 5
+
+  echo "Creating X.509 user..."
+  ./mongosh --port $port \
+    --eval 'db.getSiblingDB("$external").runCommand({
+      createUser: "OU=eng,O=mongodb,L=NY,ST=NY,C=US",
+      roles: [
+        { role: "readWrite", db: "test" },
+        { role: "userAdminAnyDatabase", db: "admin" }
+      ]
+    })'
+
+  echo "Stopping MongoDB to restart with auth..."
+  $mongod_dir/bin/mongod --dbpath $db_path --shutdown
+  sleep 5
+
+  echo "Starting MongoDB with X.509 authentication enabled..."
+  $mongod_dir/bin/mongod \
+    --dbpath $db_path \
+    --port $port \
+    --tlsMode requireTLS \
+    --tlsCertificateKeyFile "$RESOURCES_DIR/authentication_test/X509/server.pem" \
+    --tlsCAFile "$RESOURCES_DIR/authentication_test/X509/ca.crt" \
+    --bind_ip localhost &
+
+  echo "MongoDB started with X.509 authentication enabled on port $port"
+}
```

Comment:
I asked this question in slack, but should we start capturing these configurations (and data, like users and information) in config files?

Reply:
That's a good idea, this is an area we can improve also.  We can chat about it during our on-site brainstorming.  I'll create a ticket.